src/crypto/CHIPCryptoPAL.h

```diff
@@ -1475,6 +1479,38 @@ class SymmetricKeyContext
     virtual void Release() = 0;
 };
 
+/**
+ *  @brief P256 key builder for Operatinal and Ephermal keys.
+ **/
+class P256KeypairBuilder
```

Comment:
What about the following situation:

- A fail-safe is armed
- A new keypair is generated by a CSRRequest
- UpdateNOC is done
- Somehow, prior to CommissioningComplete, fail-safe expires. The fabric has to revert to using the prior key.

However, based on the usage here, the key will have been "committed" immediately upon building.

- Is there a way, with the API, to "commit" a key only when ready, to a slot, from an ephemeral key?
- Otherwise, is there a way to revert a key (probably not)

Reply:
@sujaygkulkarni-nxp ?

Reply:
@tcarmelveilleux Added the place holders for the required functionality. Could you please review the same.